api/nvidia/v1/clusterpolicy_types.go

```diff
@@ -849,6 +851,104 @@ type SandboxDevicePluginSpec struct {
+type DRADriverGPUs struct {
+	// Enabled indicates if GPUs are enabled in the NVIDIA DRA Driver
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Enable GPUs in the NVIDIA DRA Driver"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch"
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// KubeletPlugin defines configuration for the NVIDIA DRA Driver kubelet plugin
+	KubeletPlugin DRADriverKubeletPlugin `json:"kubeletPlugin,omitempty"`
+}
+
+// DRADriverComputeDomains defines configuration for ComputeDomains in the NVIDIA DRA Driver
+type DRADriverComputeDomains struct {
+	// Enabled indicates if ComputeDomains are enabled in the NVIDIA DRA Driver
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Enable ComputeDomains in the NVIDIA DRA Driver"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:booleanSwitch"
+	Enabled *bool `json:"enabled,omitempty"`
+
+	// Controller defines configuration for the NVIDIA DRA Driver controller
+	Controller DRADriverController `json:"controller,omitempty"`
+
+	// KubeletPlugin defines configuration for the NVIDIA DRA Driver kubelet plugin
+	KubeletPlugin DRADriverKubeletPlugin `json:"kubeletPlugin,omitempty"`
+}
+
+// DRADriverController defines configuration for the NVIDIA DRA Driver controller
+type DRADriverController struct {
+	// Optional: List of environment variables
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Environment Variables"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:advanced,urn:alm:descriptor:com.tectonic.ui:text"
+	Env []EnvVar `json:"env,omitempty"`
+
+	// Optional: Define resources requests and limits
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Resource Requirements"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:advanced,urn:alm:descriptor:com.tectonic.ui:resourceRequirements"
+	Resources *ResourceRequirements `json:"resources,omitempty"`
+
+	// Optional: Set tolerations
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Tolerations"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:advanced,urn:alm:descriptor:io.kubernetes:Tolerations"
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
```

Comment:
why Tolerations?

Reply:
We add tolerations to every component managed by the gpu-operator. It's common for gpu-operator operands to run on nodes with specific taints, so we need to support setting the corresponding tolerations to the operands